service-question-v1/src/questions/question.repository.ts

```diff
@@ -1,38 +1,36 @@
-import { Injectable } from "@nestjs/common";
-import { InjectModel } from "@nestjs/mongoose";
-import { Model } from "mongoose";
-import { QuestionDto } from "./question.model";
-import { Question, QuestionDocument } from "./question.schema";
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { QuestionDto } from './question.model';
+import { Question, QuestionDocument } from './question.schema';
 
 export interface QuestionRepository {
-    getAllQuestions(): Promise<QuestionDto[]>;
-    getQuestionById(questionId: string);
-    addQuestion(questionDto: QuestionDto);
-    deleteQuestion(questionId: string);
-    editQuestion(questionId: string, questionDto: QuestionDto); 
+  getAllQuestions(): Promise<QuestionDto[]>;
+  getQuestionById(questionId: string);
+  addQuestion(questionDto: QuestionDto);
+  deleteQuestion(questionId: string);
+  editQuestion(questionId: string, questionDto: QuestionDto);
 }
 
-
 @Injectable()
 export class QuestionMongoRepository implements QuestionRepository {
+  constructor(
+    @InjectModel(Question.name) private questionModel: Model<QuestionDocument>,
+  ) {}
 
-    constructor(@InjectModel(Question.name) private questionModel: Model<QuestionDocument>){}
-
-
-    async getAllQuestions(): Promise<QuestionDto[]> {
-        return await this.questionModel.find().exec();
-    }
-    async getQuestionById(questionId: string) {
-        return await this.questionModel.findById(questionId);
-    }
-    async addQuestion(questionDto: QuestionDto) {
-        this.questionModel.create(questionDto);
-    }
-    async deleteQuestion(questionId: string) {
-        await this.questionModel.findByIdAndDelete(questionId);
-    }
-    async editQuestion(questionId: string, questionDto: QuestionDto) {
-        await this.questionModel.findByIdAndUpdate(questionId, questionDto);
-    }
-
-}
\ No newline at end of file
+  async getAllQuestions(): Promise<QuestionDto[]> {
+    return await this.questionModel.find().exec();
+  }
+  async getQuestionById(questionId: string) {
+    return await this.questionModel.findById(questionId);
+  }
+  async addQuestion(questionDto: QuestionDto) {
+    this.questionModel.create(questionDto);
+  }
+  async deleteQuestion(questionId: string) {
+    await this.questionModel.findByIdAndDelete(questionId);
+  }
+  async editQuestion(questionId: string, questionDto: QuestionDto) {
+    await this.questionModel.findByIdAndUpdate(questionId, questionDto);
+  }
```

Comment:
We don't have to return Promise here, we can let TS infer the type from the return statement.

Mongoose returns a richer type that extends the Promise interface, so we get the classic `then`, `catch`, `finally` with async/await capabilities, but we also get all the extra methods from the Query interface too.

Also we don't actually have to use async/await here, we only do that when we have to unwrap the Promise (which is typically in the frontend).